code/script.js

```diff
@@ -243,101 +279,198 @@ const generateBoard = () => {
-      )
+      customPopup(`Yes, the person wears ${attribute}! Keep all that wears ${attribute}`, false)
     } else {
-      alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
-      )
+      customPopup(`No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`, false)
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
+  } 
+   else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      // customPopup popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      customPopup(
+        `Yes, the person has ${group} ${physicalAttribute}! Keep all persons with ${group} ${physicalAttribute}`, false
+      )
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      // customPopup popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      customPopup(
+        `No, the person has not ${group} ${physicalAttribute}! Remove all persons with ${group} ${physicalAttribute}`, false
+      )
     }
+
   }
 
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
+  charactersInPlay = charactersInPlay.filter((person) => {
+    if (keep) {
+      return person[attribute] === physicalAttribute
+    }
+    else {
+      return person[attribute] != physicalAttribute
 
+    }
+  })
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
+// store the interaction from the player in a variable.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
   // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  customPopup(`Are you sure you want to guess on ${suspect}?`, true, suspect)
 }
 
 // If you confirm, this function is invoked
+// 1. Check if the suspect is the same as the secret person's name
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
+
   // 2. Set a Message to show in the win or lose section accordingly
   // 3. Show the win or lose section
+  winLose.classList.add('win-or-lose-wrapper-on')
+  if (suspect === secret.name) {
+    resulText.textContent = 'Congrats! You win'
+  } else {
+    resulText.textContent = 'GAME OVER! You Lose 😖'
+  }
   // 4. Hide the game board
 }
+//Bonus Step
+const playAgain = () => {
+  start()
+  winLose.classList.remove('win-or-lose-wrapper-on')
```

Comment:
Really liked that you used the classList.add/remove and the text.Content property🙌  I didn't know that the last one existed. Cool! 

I noticed that the win-or-lose-wrapper didn't cover all of the gameboard, just the viewheight (vh), so when you scroll the page the board is visible. I don't know if that is your intention or not, just a food for thought🧐